src/core/pattern.js

```diff
@@ -186,13 +185,62 @@ class RadialAxialShading extends BaseShading {
     const color = new Float32Array(cs.numComps),
       ratio = new Float32Array(1);
     let rgbColor;
-    for (let i = 0; i <= NUMBER_OF_SAMPLES; i++) {
+
+    let iBase = 0;
+    ratio[0] = t0;
+    fn(ratio, 0, color, 0);
+    let rgbColorBase = cs.getRgb(color, 0);
+    const cssColorBase = Util.makeHexColor(
+      rgbColorBase[0],
+      rgbColorBase[1],
+      rgbColorBase[2]
+    );
+    colorStops.push([0, cssColorBase]);
+
+    let iPrev = 1;
+    ratio[0] = t0 + step;
+    fn(ratio, 0, color, 0);
+    let rgbColorPrev = cs.getRgb(color, 0);
+
+    for (let i = 2; i < NUMBER_OF_SAMPLES; i++) {
       ratio[0] = t0 + i * step;
       fn(ratio, 0, color, 0);
       rgbColor = cs.getRgb(color, 0);
-      const cssColor = Util.makeHexColor(rgbColor[0], rgbColor[1], rgbColor[2]);
-      colorStops.push([i / NUMBER_OF_SAMPLES, cssColor]);
+
+      // See if rgbColorPrev is predicted by lerping rgbColorBase and rgbColor.
```

Comment:
I had to lookup what "lerping" even meant, so please just spell it out save time for future readers.

Reply:
Ah, you've caught me in a comment I wrote to remind me what needed to be done. Polished up a bit to be more consistent with how the code turned out.